hibernate-core/src/main/java/org/hibernate/engine/transaction/jta/platform/internal/WebSphereJtaPlatform.java

```diff
@@ -46,7 +48,7 @@ public WebSphereJtaPlatform() {
 			throw new JtaPlatformException( "Could not locate WebSphere TransactionManager access class" );
 		}
 
-		this.transactionManagerAccessClass = tmAccessClass;
+		this.transactionManagerAccessClass = NullnessUtil.castNonNull( tmAccessClass );
```

Comment:
I think it might be better to replace

`private final Class transactionManagerAccessClass;`

with

`private final Method getTransactionManagerMethod;`

and this here with `this.getTransactionManagerMethod = NullnessUtil.castNonNull( NullnessUtil.castNonNull( tmAccessClass ).getMethod( "getTransactionManager" ) )`

This way, you should be able to avoid the `@SuppressWarnings("nullness")` in `locateTransactionManager`

Reply:
Removing that `@SuppressWarnings("nullness")` in locateTransactionManager doesn't need anything really, I just forgot to remove it, since it was combined with another warning I overlooked it.